pkg/inframetadata/internal/hostmap/hostmap_test.go

```diff
@@ -0,0 +1,139 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package hostmap
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	conventions "go.opentelemetry.io/collector/semconv/v1.18.0"
+
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/inframetadata/internal/testutils"
+	"github.com/DataDog/opentelemetry-mapping-go/pkg/inframetadata/payload"
+)
+
+func TestUpdate(t *testing.T) {
+	hostInfo := []struct {
+		hostname        string
+		attributes      map[string]any
+		expectedChanged bool
+		expectedErr     string
+	}{
+		{
+			hostname: "host-1-hostid",
+			attributes: map[string]any{
+				conventions.AttributeCloudProvider: conventions.AttributeCloudProviderAWS,
+				conventions.AttributeHostID:        "host-1-hostid",
+				conventions.AttributeHostName:      "host-1-hostname",
+				conventions.AttributeOSDescription: "Fedora Linux",
+				conventions.AttributeHostArch:      conventions.AttributeHostArchAMD64,
+				attributeKernelName:                "GNU/Linux",
+				attributeKernelRelease:             "5.19.0-43-generic",
+				attributeKernelVersion:             "#44~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon May 22 13:39:36 UTC 2",
+			},
+			expectedChanged: false,
+		},
+		{
+			// Same as #1, but missing some attributes
+			hostname: "host-1-hostid",
+			attributes: map[string]any{
+				conventions.AttributeCloudProvider: conventions.AttributeCloudProviderAWS,
+				conventions.AttributeHostID:        "host-1-hostid",
+				conventions.AttributeHostName:      "host-1-hostname",
+				conventions.AttributeOSDescription: "Fedora Linux",
+			},
+			expectedChanged: false,
+		},
+		{
+			// Same as #1 but wrong type and an update
+			hostname: "host-1-hostid",
+			attributes: map[string]any{
+				conventions.AttributeCloudProvider: conventions.AttributeCloudProviderAWS,
+				conventions.AttributeHostID:        "host-1-hostid",
+				conventions.AttributeHostName:      "host-1-hostname",
+				conventions.AttributeOSDescription: true, // wrong type
+				conventions.AttributeHostArch:      conventions.AttributeHostArchAMD64,
+				attributeKernelName:                "GNU/Linux",
+				attributeKernelRelease:             "5.19.0-43-generic",
+				attributeKernelVersion:             "#82~18.04.1-Ubuntu SMP Fri Apr 16 15:10:02 UTC 2021", // changed
+			},
+			expectedChanged: true,
+			expectedErr:     "\"os.description\" has type \"Bool\", expected type \"Str\" instead",
```

Comment:
nit: consider adding a test with multiple errors that are all reported

Reply:
Done